app/src/main/java/com/mycompany/myapp/ui/main/MainPresenter.java

```diff
@@ -62,11 +74,74 @@ public String getAuthor() {
         }
     }
 
-    @Parcel
-    public static class State {
-        String username = "madebyatomicrobot";
-        String repository = "android-starter-project";
-        List<CommitViewModel> commits = new ArrayList<>();
+    @AutoParcelGson
+    public abstract static class MainViewModel {
+        static Builder builder() {
+            return new AutoParcelGson_MainPresenter_MainViewModel.Builder()
+                    .username("")
+                    .repository("")
+                    .fingerprint(String.format("Fingerprint: %s", BuildConfig.VERSION_FINGERPRINT))
+                    .version(String.format("Version: %s", BuildConfig.VERSION_NAME))
+                    .commits(Collections.emptyList());
+
+        }
+
+        abstract String username();
+
+        abstract String repository();
+
+        abstract String fingerprint();
+
+        abstract String version();
+
+        abstract List<CommitViewModel> commits();
+
+        abstract Builder toBuilder();
+
+        @AutoParcelGson.Builder
+        abstract static class Builder {
+            abstract Builder username(String username);
+
+            abstract Builder repository(String repository);
+
+            abstract Builder fingerprint(String fingerprint);
+
+            abstract Builder version(String version);
+
+            abstract Builder commits(List<CommitViewModel> commits);
```

Comment:
Should CommitViewModel be autovalued so you can get the .equals() correct behavior?